site/content/en/docs/build-clusters.md

```diff
@@ -0,0 +1,181 @@
+default one. For example, [Google's OSS Prow instance](https://oss.gprow.dev)
+has its own script
+[here](https://github.com/GoogleCloudPlatform/oss-test-infra/blob/master/prow/oss/create-build-cluster.sh).
+
+The scripts all have prompts and ask you various questions to set everything up.
+If everything proceeds smoothly, then there's nothing more for you to do (you're
+ready to start writing Prow jobs that use your cluster). Below is a discussion
+of the overall process to demystify what goes on behind the scenes.
+
+## How does Prow actually use your build cluster?
+
+There are two requirements:
+
+1. Prow must be able to schedule jobs into your build cluster.
+2. The jobs themselves must be able to upload artifacts to the GCS bucket used
+by [Deck](../components/deck), in order to report job status (e.g., "passing" or
+"failing").
+
+We look at both requirements below.
+
+### Let Prow schedule your jobs into your build cluster
+
+Prow is a Kubernetes cluster. So is your build cluster. In order for Prow to
+schedule jobs (i.e., create Kubernetes pods) into your build cluster, it must be
+able to authenticate as a KSA defined in your build cluster that has a
+[cluster-admin](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles)
+Kubernetes role. This way, Prow (with its various individual components) can
+freely create, update, and delete jobs (pods) in your build cluster as
+necessary. The item "KSA A" in the diagram below is this service account.
+
+In order for Prow's components to authenticate as the cluster-admin KSA in your
+build cluster, they use a token. You can think of this token simply as a
+password, that when provided to your build cluster's API server along with the
+KSA name, grants authentication as this very same KSA.
+
+The question now turns to how we can generate this token. This can be done
+manually, but the token cannot be valid for longer than 2 days for security
+reasons (a GKE restriction), and must be rotated regularly. Fortunately, there
+is a tool called `gencred` that automates the generation of the token. We just
+need to have it be run periodically --- and so we need to add a Prow job that
+regularly invokes `gencred`.
+
+Once the token is generated, we can store it in the GCP Secret Manager for your
+build cluster's GCP project for safekeeping. Then we have to bring this token
+into Prow and mount it into the various Prow components that need it; this
+one-way sync is performed by the `kubernetes-external-secrets` component. The
+Prow components' configurations don't have to be updated though, because your
+build cluster's token is combined with other secrets into a composite file.
+
+### Let your jobs report their status to GCS
+
+Your jobs in your build cluster must have GCS access in order to upload critical
+job metadata, such as a `finished.json` file to indicate the status of your job
+(whether it passed or failed). The GCS bucket location depends on how the Prow
```

Comment:
the GCS bucket could be configured on several different levels in Prow, and is not the same for a single Prow instance, for example a user can configure bucket for a build cluster (https://github.com/GoogleCloudPlatform/oss-test-infra/blob/d46723826c93a03f1d6e30d603bd5d0b0f723212/prow/oss/config.yaml#L138). Rarely user might want to use a different bucket for a job, which is also allowed

Reply:
done